tests/unit/test_price_reduction.py

```
@@ -0,0 +1,431 @@
+
+
+@runtime_checkable
+class _ApplyAutoPriceReduction(Protocol):
+    def __call__(self, ad_cfg:SimpleNamespace, ad_cfg_orig:dict[str, Any], ad_file_relative:str) -> None:
+        ...
+
+
+@pytest.fixture
+def apply_auto_price_reduction() -> _ApplyAutoPriceReduction:
+    bot = KleinanzeigenBot()
+    return cast(_ApplyAutoPriceReduction, getattr(bot, "_KleinanzeigenBot__apply_auto_price_reduction"))
```

Ruff (0.14.4)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)